trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/model/ConfigurationQueryParameters.java

```diff
@@ -26,6 +26,9 @@ public interface ConfigurationQueryParameters {
      * @return The query parameter map.
      */
     @NonNull
-    @Schema(required = true, description = "Parameters as specified in corresponding ConfigurationTypeDescriptor. Use key `path` for file URI string values.")
+    @Schema(required = true, description = "Parameters to create a configuration. Use optional key `name` for unique name of the configuration. "
+            + "Use optional key `description` to provide a description of the configuration. Use optional key `typeId` to specify the configuration source type. "
+            + "The typeId is not required if endpoint URI contains the typeId. Use key `properties` to specify JSON parameters according the JSON schema of"
+            + "the corresponding ConfigurationTypeDescriptor. If no JSON schema is used, add the properties directly as key value pairs. Use key `path` for file URI string values.")
```

Comment:
I understand. Using something RequestedQueryParameters and I tried it. The issues I had is the different ways we supply the custom properties:

1. through JSON schema. This requires the properties key.
2. directly under parameters (e.g. path for custom XML analysis in global configuration)

I had the following. It works well with 1). Do you know how can we support case 2)?
```java
public interface ConfigurationQueryParameters {

    /**
     * @return The parameters.
     */
    @NonNull
    @Schema(required = true)
    ConfigurationParameters getParameters();

    /**
     * No expected properties below, as per current trace-server protocol.
     */
    interface ConfigurationParameters {
        @JsonProperty("name")
        @Schema(description = "Optional, unique name of the configuration. If omitted a unique name will be generated.")
        String getName();
        @JsonProperty("description")
        @Schema(description = "Optional, description of the configuration.")
        String getDescription();
        @Schema(description = "Optional, typeId of the configuration. Omit if it's part of endpoint URI.")
        @JsonProperty("typeId")
        String getTypeId();
        @JsonProperty("properties")
        @Schema(description = "Properties as JSON object as specified in the schema of the corresponding ConfigurationTypeDescriptor. Use key `path` for file URI string values.")
        Map<String, Object> getProperties();
    }
}
```

Reply:
The 'properties' could also be used for global configuration?
It could be as specified in a schema or in a list of paramDescriptors.
We should set the 'required' value for each parameter.

Reply:
> The 'properties' could also be used for global configuration? It could be as specified in a schema or in a list of paramDescriptors. We should set the 'required' value for each parameter.

Sounds good. Thanks for the feedback.

Reply:
Done